packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -2731,12 +2778,15 @@ class _DayItemState extends State<_DayItem> {
         style: _HighlightPainterStyle.highlightAll,
         textDirection: textDirection,
       );
+      if (widget.isDisabled) {
```

Comment:
Not necessarily related to this PR: All the `if...else if` blocks feel confusing and repetitive. Might be worth it to refactor to use switch statements.

Reply:
In previous feedbacks, I've been asked to only focus on the feature, and here you ask me to change something unrelated. I'm not sure if it's the good PR to do that 😅

Reply:
> In previous feedbacks, I've been asked to only focus on the feature, and here you ask me to change something unrelated. I'm not sure if it's the good PR to do that 😅

I'm not sure if I clarified, but not in this PR, it's just a general point.